src/index.ts

```diff
@@ -153,11 +153,15 @@ const operatingSystemRules: OperatingSystemRule[] = [
   ['Search Bot', SEARCHBOT_OS_REGEX],
 ];
 
-export function detect(): BrowserInfo | BotInfo | NodeInfo | null {
+export function detect(userAgent: string | null = null): BrowserInfo | BotInfo | NodeInfo | null {
   if (typeof navigator !== 'undefined') {
     return parseUserAgent(navigator.userAgent);
   }
 
+  if (userAgent) {
```

Comment:
I think we should push this above the `navigator` check. I think if someone has gone to the effort of specifying the `userAgent` argument that should take priority over everything else.

Additionally, I think being specific in the `if` clause is probably worthwhile, e.g.:

```ts
if (userAgent !== undefined) {
  ...
}
```

Reply:
Yeah, think it's better to place the condition before the `navigator` check. Thanks for an advice :)